src/spring/azext_spring/jobs/job.py

```diff
@@ -530,3 +536,39 @@ def _get_default_writer():
     Define this method, so that we can mock this method in scenario test to test output
     """
     return DefaultWriter()
+
+
+def _job_log_stream(cmd, client, resource_group, service, name, execution, all_instances=None, instance=None,
+                    follow=None, max_log_requests=5, lines=100, since=None, limit=2048):
+    queryOptions = LogStreamBaseQueryOptions(follow=follow, lines=lines, since=since, limit=limit)
+    url_dict = _get_log_stream_urls(cmd, client, resource_group, service, name, execution, all_instances,
+                                    instance, queryOptions)
+    validate_thread_number(follow, len(url_dict), max_log_requests)
+    auth = get_bearer_auth(cmd.cli_ctx)
+    exceptions = []
+    threads = _get_log_threads(all_instances, url_dict, auth, exceptions)
+
+    if follow and len(threads) > 1:
+        parallel_start_threads(threads)
+    else:
+        sequential_start_threads(threads)
+
+    if exceptions:
+        raise exceptions[0]
+
+
+def _handle_log_stream_permission_deny(cmd, resource_group, service, operation_name):
+    sub_id = get_subscription_id(cmd.cli_ctx)
+    resource_id = get_service_instance_resource_id(sub_id=sub_id, group=resource_group, service=service)
+    msg = f"(AuthorizationFailed) You do not have authorization to {operation_name} over the scope '{resource_id}' . " \
+          f"Please check if you have the Azure role '{JOB_LOG_READER_ROLE_NAME}' ." \
+          " If access was recently granted, please refresh your credentials."
+    raise CLIError(msg)
```

Comment:
Please use the more specific error type `PermissionDenyError`

Reply:
Updated. Waiting for pull request gate pipelines.